desktop/core/src/desktop/js/utils/hooks/useQueueProcessor.ts

```diff
@@ -16,63 +16,61 @@
 
 import { useState, useEffect } from 'react';
 
-type QueueProcessorState<T> = {
-  pendingQueue: T[];
-  processingQueue: T[];
-  completedQueue: T[];
-};
-
-type UseQueueProcessorResult<T> = QueueProcessorState<T> & {
+interface UseQueueProcessorResult<T> {
+  queue: T[];
   enqueue: (newItems: T[]) => void;
   dequeue: (item: T) => void;
-  isProcessing: boolean;
-};
+  isLoading: boolean;
+}
+
+interface UseQueueProcessorOptions {
+  concurrentProcess: number;
+  onSuccess?: () => void;
+}
 
 const useQueueProcessor = <T>(
   onItemProcess: (item: T) => Promise<void>,
-  concurrentProcess: number = 1
+  options: UseQueueProcessorOptions
 ): UseQueueProcessorResult<T> => {
-  const [pendingQueue, setPendingQueue] = useState<QueueProcessorState<T>['pendingQueue']>([]);
-  const [processingQueue, setProcessingQueue] = useState<QueueProcessorState<T>['processingQueue']>(
-    []
-  );
-  const [completedQueue, setCompletedQueue] = useState<QueueProcessorState<T>['completedQueue']>(
-    []
-  );
-
-  const processQueueItem = async () => {
-    const nextItem = pendingQueue.shift();
-
-    if (nextItem) {
-      setProcessingQueue(prev => [...prev, nextItem]);
-      await onItemProcess(nextItem);
-      setProcessingQueue(prev => prev.filter(i => i !== nextItem));
-      setCompletedQueue(prev => [...prev, nextItem]);
-      processQueueItem();
-    }
-  };
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [queue, setQueue] = useState<T[]>([]);
+  const [processingQueue, setProcessingQueue] = useState<T[]>([]);
 
   const enqueue = (newItems: T[]) => {
-    setPendingQueue(prevQueue => [...prevQueue, ...newItems]);
+    setQueue(prevQueue => [...prevQueue, ...newItems]);
   };
 
   const dequeue = (item: T) => {
-    setPendingQueue(prev => prev.filter(i => i !== item));
+    setQueue(prev => prev.filter(i => i !== item));
```

Comment:
nit: what if the item is of a complex data type and that data type has been recreated before passed to the dequeue (like in the unit tests)? You could add a comparison function to the dequeue function input to support such cases... 

I'm only being this picky since you are ambitious and creating a nice reusable hook ;-)

Reply:
Great suggestion, I completely agree on comparison function. Let me take this in separate PR.
This will also solve the previous comment on checking by reference.